src/CarbonSystemSolver.jl

```diff
@@ -0,0 +1,1750 @@
+@inline Sᴾ⁰⁵(Sᵖ)  = sqrt(Sᵖ)
+@inline ΘᵒC(Θᴷ)   = Θᴷ - Θᴷ_0ᵒC
+@inline ΔSᵖ(Sᵖ)   = Sᵖ - 34.8
+@inline Rₜ(Θᴷ)     = gasconst_bar_cm3_o_mol_k * Θᴷ
+
+@inline """
+    H₂Oˢʷ(Sᵖ)
+
+    # Function returns the mass of pure water in one kg of seawater
+    # of salinity s
+    # References: "libthdyct" -- derived by Munhoven (1997) from data by Millero (1982)
+    #             "Handbook (2007)" -- Handbook (2007)
+    # pH scale:   N/A
+"""
+H₂Oˢʷ(Sᵖ) =  1. - 0.001005*Sᵖ 
+#H₂Oˢʷ(Sᵖ) = 1. - 0.0010049*Sᵖ # libthdyct
+
+@inline """
+    μₛ(Sᵖ)
+
+    # Function calculates ionic strength in mol/kg-SW, for given salinity.
+    # References: "libthdyct" -- derived by Munhoven (1997) from data by Millero (1982)
+    #             "Handbook (2007)" -- Handbook (2007)
+    # pH scale:   N/A
+"""
+μₛ(Sᵖ)    = (0.019924 * Sᵖ) / H₂Oˢʷ(Sᵖ) # Handbook (2007)
+# μₛ(Sᵖ)    = (0.019920 * Sᵖ) / H₂Oˢʷ(Sᵖ)# libthdyct
+
+@inline """
+    Bᵀᴼᵀ(Sᵖ)
+
+    # Total borate concentration in mol/kg-SW given the salinity of a sample
+    # References: Uppström (1974), cited by  Dickson et al. (2007, chapter 5, p 10)
+    #             Millero (1982) cited in Millero (1995)
+"""
+Bᵀᴼᵀ(Sᵖ)  = 0.000416*(Sᵖ/35.)
+#Bᵀᴼᵀ(Sᵖ)  = 0.000232 * (Sᵖ/1.80655)/10.811
+
+@inline """
+    Caᵀᴼᵀ(Sᵖ)
+
+    # Total calcium concentration in mol/kg-SW given the salinity of a sample
+    # References: Culkin and Cox (1966), Culkin (1967), Riley and Tongudai (1967)    
+"""
+Caᵀᴼᵀ(Sᵖ) = (0.02127/40.078) * (Sᵖ/1.80655)
+# Caᵀᴼᵀ(Sᵖ) = 0.010282*(Sᵖ/35.)
+
+@inline """
+    Fᵀᴼᵀ(Sᵖ)
+
+    # Total fluoride concentration in mol/kg-SW given the salinity of a sample
+    # References: Culkin (1965) (???)
+"""
+Fᵀᴼᵀ(Sᵖ)  = 0.000068*(Sᵖ/35.)
+
+@inline """
+    SO₄ᵀᴼᵀ(Sᵖ)
+
+    # Total sulfate concentration in mol/kg-SW given the salinity of a sample
+    # References: Morris, A.W. and Riley, J.P. (1966) quoted in Handbook (2007)
+"""
+SO₄ᵀᴼᵀ(Sᵖ) = (0.1400/96.062)*(Sᵖ/1.80655)
+#SO₄ᵀᴼᵀ(Sᵖ) = 0.028234*(Sᵖ/35.)
+
+@inline """
+    NH₄ᵀᴼᵀ(Sᵖ)
+
+    # Total ammonium concentration in mol/kg-SW given the salinity of a sample
+    # References: Yamamoto (1995)
+"""
+NH₄ᵀᴼᵀ(Sᵖ) = (0.0001/17.031) * (Sᵖ/1.80655)
+
+@inline """
+    PO₄ᵀᴼᵀ(Sᵖ)
+
+    # Total phosphate concentration in mol/kg-SW given the salinity of a sample
+    # References: Millero (1995)
+"""
+PO₄ᵀᴼᵀ(Sᵖ) = (0.0001/94.971) * (Sᵖ/1.80655)
+
+@inline """
+    SiO₃ᵀᴼᵀ(Sᵖ)
+
+    # Total silicate concentration in mol/kg-SW given the salinity of a sample
+    # References: Millero (1995)
+"""
+SiO₃ᵀᴼᵀ(Sᵖ) = (0.0001/60.084) * (Sᵖ/1.80655)
+
+@inline """
+    H₂Sᵀᴼᵀ(Sᵖ)
+
+    # Total hydrogen sulfide concentration in mol/kg-SW given the salinity of a sample
+    # References: Dickson (1990)
+"""
+H₂Sᵀᴼᵀ(Sᵖ) = (0.0001/34.082) * (Sᵖ/1.80655)
+
+##=======================================================================
+## AK_CARB_0_WEIS74 <- function (Θᴷ, s)
+##=======================================================================
+const Pᵈⁱᶜₖ₀ = (
```

Comment:
I think we should use `struct` rather than constant `NamedTuple`, what do you think? `struct` allows the parameters to be changed without any downsides, I don't think.